target/linux/ramips/dts/mt7620a_dlink_dap-1620-a2.dts

```diff
@@ -0,0 +1,204 @@
+			linux,code = <KEY_RESTART>;
+			debounce-interval = <60>;
+		};
+
+		wps {
+			label = "wps";
+			gpios = <&gpio0 2 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_status_red: status_red {
+			function = LED_FUNCTION_STATUS;
+			color = <LED_COLOR_ID_RED>;
+			gpios = <&gpio0 9 GPIO_ACTIVE_LOW>;
+		};
+
+		led_status_green: status_green {
+			function = LED_FUNCTION_STATUS;
+			color = <LED_COLOR_ID_GREEN>;
+			gpios = <&gpio0 10 GPIO_ACTIVE_LOW>;
+		};
+
+		rssi_low_red {
+			color = <LED_COLOR_ID_RED>;
+			function = LED_FUNCTION_INDICATOR;
+			function-enumerator = <1>;
+			gpios = <&gpio0 13 GPIO_ACTIVE_LOW>;
+		};
+
+		rssi_low_green {
+			color = <LED_COLOR_ID_GREEN>;
+			function = LED_FUNCTION_INDICATOR;
+			function-enumerator = <1>;
+			gpios = <&gpio0 12 GPIO_ACTIVE_LOW>;
+		};
+
+		rssi_med_green {
+			color = <LED_COLOR_ID_GREEN>;
+			function = LED_FUNCTION_INDICATOR;
+			function-enumerator = <2>;
+			gpios = <&gpio0 8 GPIO_ACTIVE_LOW>;
+		};
+
+		rssi_high_green {
+			color = <LED_COLOR_ID_GREEN>;
+			function = LED_FUNCTION_INDICATOR;
+			function-enumerator = <3>;
+			gpios = <&gpio0 11 GPIO_ACTIVE_LOW>;
+		};
+	};
+};
+
+&spi0 {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <50000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0 0x30000>;
+				read-only;
+			};
+
+			partition@30000 {
+				label = "nvram";
+				reg = <0x30000 0x10000>;
+			};
+
+			partition@40000 {
+				label = "factory";
+				reg = <0x40000 0x10000>;
+				read-only;
+
+				nvmem-layout {
+					compatible = "fixed-layout";
+					#address-cells = <1>;
+					#size-cells = <1>;
+
+					eeprom_factory_0: eeprom@0 {
+						reg = <0x0 0x200>;
+					};
+
+					eeprom_factory_8000: eeprom@8000 {
+						reg = <0x8000 0x200>;
+					};
+
+					macaddr_factory_ffd4: macaddr@ffd4 {
+						compatible = "mac-base";
+						reg = <0xffd4 0x11>;
+						#nvmem-cell-cells = <1>;
+					};
+
+					macaddr_factory_ffe8: macaddr@ffe8 {
+						compatible = "mac-base";
+						reg = <0xffe8 0x11>;
+						#nvmem-cell-cells = <1>;
+					};
+				};
+			};
+
+			partition@50000 {
+				compatible = "denx,uimage";
+				label = "firmware";
+				reg = <0x50000 0x7b0000>;
+			};
+		};
+	};
+};
+
+&state_default {
+	gpio {
+		groups = "i2c", "uartf";
+		function = "gpio";
+	};
+};
+
+&ethernet {
+	pinctrl-names = "default";
+	pinctrl-0 = <&rgmii1_pins &mdio_pins>;
+
+	nvmem-cells = <&macaddr_factory_ffe8 0>;
+	nvmem-cell-names = "mac-address";
+
+	mediatek,portmap = "wllll";
+
+	port@5 {
+		status = "okay";
+		phy-handle = <&phy1>;
+		phy-mode = "rgmii";
+	};
+
+	mdio-bus {
+		status = "okay";
+
+		phy1: ethernet-phy@1 {
+			reg = <1>;
+			phy-mode = "rgmii";
+		};
+
+		phy1f: ethernet-phy@1f {
+			reg = <0x1f>;
+			phy-mode = "rgmii";
+		};
```

Comment:
What do you need this PHY definition for? it is also not referenced.

If this is not needed you can delete it.

Reply:
Unfortunately, this is the only way I can get the Power LED light up green post booting. It doesn't respond to any other GPIO pin commands. Not sure if this call code triggers any bus which powers the LED.

The GPIO 10 (Green-Power) always shows resource busy and I'm unable to toggle it with any command or UI.

Reply:
> this is the only way I can get the Power LED light up green post booting.

This should be commented on in the dts file.
Also, the `phy1f` label can be dropped even if the dt node is needed.

Reply:
> Unfortunately, this is the only way I can get the Power LED light up green post booting. It doesn't respond to any other GPIO pin commands.

This sounds curious indeed, so it would only affect `status_green` (above wps button) but not `status_red`? Maybe there are indeed extra PCB traces between them, e.g. if the manufacturer wanted to reserve the option to make one of the LEDs hardware-controllable from an external phy, which would provide the 3.3v (so they could make it flash (i.e. turn off) by either pulling to 3v3 on the MCU side, or by dropping the supply voltage on the PHY GPIO to GND).

`resource busy` usually means the LED is controlled by the kernel, i.e. after you have declared a function in the dts, you can no longer control it low-level from userspace using /sys/class/gpio calls, so this would be normal.

Reply:
That's what I thought as well thinking its hard-wired.
- As soon the device is switched ON, green power LED comes up.
- Post kernel load without phyif the green switches OFF. With phyif the green stays ON.
- Red is working as it should like other GPIO LEDs. Its now configured against the "led-boot", so effectively the device blinks orange (Green+Red) during post-kernel-boot and once completely booted and when the eth0 comes up, red goes OFF and device shows green indicating the Power ON state.

At one point I was thinking just use Red LED for power indication, forgetting green all together, but the "red" makes it look odd and also glaringly bright compared to green when functioning normally.